pubsub/README.md

```diff
@@ -149,22 +156,172 @@ economics (see e.g.
+* `Lax` (legacy, insecure, underterministic, to be deprecated): accept a signed
+  message if the signature verification passes, or if it's unsigned.
+
 When signature validation fails for a signed message, the implementation must
-drop the message and omit propagation. Locally, it may treat this event in whichever
-manner it wishes (e.g. logging).
+drop the message and omit propagation. Locally, it may treat this event in
+whichever manner it wishes (e.g. logging, penalization, etc.).
+
+#### Signature Policy Options
+
+The usage of the `signature`, `key`, `from`, and `seqno` fields in `Message`
+is configurable per topic.
+
+> **[[ Implementation note ]]:** At the time of writing this section,
+> go-libp2p-pubsub (reference implementation of this spec) allows for
+> configuring the signature policy at the **global pubsub instance level**.
+> This needs to be pushed down to topic-level configuration.
+> Other implementations should support topic-level configuration, as this spec
+> mandates.
+
+The intersection of signing behaviours across the two axes (signature creation
+and signature verification) gives way to four signature policy options:
+ 
+* `StrictSign`, `StrictNoSign`. Deterministic, usage encouraged.
+* `LaxSign`, `LaxNoSign`. Non-deterministic, legacy, usage discouraged. Mostly
+  for backwards compatibility. Will be deprecated. If the implementation decides
+  to support these, their use should be discouraged through deprecation warnings.
+
+**`StrictSign` option**
+
+On the producing side:
+  - Build messages with the `signature`, `key` (`from` may be enough for
+    certain inlineable public key types), `from` and `seqno` fields.
+  
+On the consuming side:
+  - Enforce the fields to be present, reject otherwise.
+  - Propagate only if the fields are valid and signature can be verified,
+    reject otherwise.
+
+**`StrictNoSign` option**
+
+On the producing side:
+  - Build messages without the `signature`, `key`, `from` and `seqno` fields.
+  - The corresponding protobuf key-value pairs are absent from the marshalled
+    message, not just empty.
+  
+On the consuming side:
+  - Enforce the fields to be absent, reject otherwise.
+  - Propagate only if the fields are absent, reject otherwise.
+  - A `message_id` function will not be able to use the above fields, and should
+    instead rely on the `data` field. A commonplace strategy is to calculate
+    a hash.
+
+**`LaxSign` legacy option**
+
+_Not required for backwards-compatibility. Considered insecure, nevertheless
+defined for completeness._
+
+Always sign, and verify incoming signatures, and but accept unsigned messages.
+
+On the producing side:
+  - Build messages with the `signature`, `key` (`from` may be enough), `from`
+    and `seqno` fields.
+  
+On the consuming side:
+  - `signature` may be absent, and not verified.
+  - Verify `signature`, iff the `signature` is present, then reject if
+    `signature` is invalid.
+
+**`LaxNoSign` option**
+
+_Previous default_.
```

Comment:
that's not quite true -- `StrictSign` has been the default in the reference implementation for some time now.

Reply:
Previous default for no-verification mode. But yes, this is confusing.

Reply:
Adjusting.